nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -77,15 +107,27 @@ class ServiceClient
       node_->get_logger(), "%s service client: send async request",
       service_name_.c_str());
     auto future_result = client_->async_send_request(request);
-
-    if (callback_group_executor_.spin_until_future_complete(future_result, timeout) !=
-      rclcpp::FutureReturnCode::SUCCESS)
-    {
-      // Pending request must be manually cleaned up if execution is interrupted or timed out
-      client_->remove_pending_request(future_result);
-      throw std::runtime_error(service_name_ + " service client: async_send_request failed");
+    if (callback_group_executor_) {
```

Comment:
This if/else should be removed and instead set the `callback_group_executor_` to the node's default executor in the constructor so that its always populated (whether its the internal executor or the `node`'s). You also need to use `spin_until_future_complete` for the same reason I mention in the BT Service Node, manually handling the futures is not functional unfortunately. I think this needs to be reverted, I think what happened before was perfect, we just needed to set the `callback_group_executor_` appropriately